plugins/aggregators/starlark/README.md

```diff
@@ -0,0 +1,91 @@
+# Starlark Aggregator
+
+The `starlark` aggregator calls a Starlark function to add the metrics to the aggregator,
+then calls another Starlark function to push the aggregates to the accumulator. Both Starlark functions
+have an argument called `cache` which is a [dict-like][dict] object allowing to keep temporary the metrics to
+aggregate. The `cache` is automatically cleared by the the aggregator.
```

Comment:
The starlark processor already has a global dict called 'state'. This serves a very similar purpose to the cache argument here. It seems like the cache argument could be removed and the aggregator starlark code could use the state dict.

https://github.com/influxdata/telegraf/blob/56398237c4a931dba438db553a1ee1487926c072/plugins/processors/starlark/README.md#L170

If we have both dicts, I think users who write both aggregators and processors may be confused by the similarity. If we can cover both use cases with the same dict, it will be more user friendly.

Reply:
I understand your point but if we do so it means that the state has to be added explicitly in all the Starlark script like [in this example] (https://github.com/influxdata/telegraf/blob/master/plugins/processors/starlark/testdata/compare_metrics.star#L10-L12). In the processor, it has been done this way as it is seen as a corner case while in case of an aggregator it is required to have a shared state/cache that's why I added it explicitly as an argument of the functions. WDYT?